lib/cloudant.js

```diff
@@ -493,14 +500,14 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   };
 
   if (id) {
-    self.updateAttributes(model, id, data, {}, function(err, docs) {
+    self.update(model, {id: id}, data, {}, function(err, docs) {
       if (err && err.statusCode !== 404) return cb(err);
       else if (err && err.statusCode === 404) {
         // strip out the `_rev` property before creation
         delete data._rev;
         self.create(model, data, {}, createHandler);
       } else {
-        return cb(err, self.fromDB(model, mo, docs), {isNewInstance: false});
+        return cb(err, docs[0], {isNewInstance: false});
```

Comment:
We do need to strip the `_rev` before creation though. If the user tries to do a create and pass in a random `_rev` value, it is upon us to strip it off before insertion.

This is not related to this PR at all. It was well discussed and accepted for the PR of [`allow _rev property`](https://github.com/strongloop/loopback-connector-cloudant/pull/123)